File: invoice-manager-ui/src/app/@core/models/cfdi/cfdi.ts
import { CfdiRelacionados } from "./cfdi-relacionados";
import { Complemento } from "./complemento";
import { Concepto } from "./concepto";
import { Emisor } from "./emisor";
import { Impuesto } from "./impuesto";
import { InformacionGlobal } from "./informacion-global";
import { Receptor } from "./receptor";

export class Cfdi{

 public id: number;

 public version: string;
 public serie: string;
 public folio: string;
 public fecha: string;
 public sello: string;
 public formaPago: string;
 public metodoPago: string;
 public noCertificado: string;
 public certificado: string;
 public condicionesDePago: string;
 public subtotal: number;
 public descuento: number;
 public total: number;
 public moneda: string;
 public tipoCambio: number;
 public tipoDeComprobante: string;
 public exportacion: string;
 public lugarExpedicion: string;
 public confirmacion: string;

 public informacionGlobal: InformacionGlobal;
 public cfdiRelacionados: CfdiRelacionados;
 public emisor: Emisor;
 public receptor: Receptor;

 public conceptos: Concepto[];
 public impuestos: Impuesto[];
 public complemento :Complemento;
 public addenda: any[];


 constructor() {
    this.version = '4.0';
    this.tipoDeComprobante = 'I';
    this.conceptos = [];
    this.emisor = new Emisor();
    this.receptor = new Receptor();
    this.complemento = new Complemento();
    this.cfdiRelacionados = new CfdiRelacionados();
    this.total = 0;
    this.subtotal = 0
    this.descuento = 0;
    this.tipoCambio = 0;
    this.serie = '1';//should be generated
}
}

File: invoice-manager-models/src/main/java/com/business/unknow/model/cfdi/Cfdi.java
package com.business.unknow.model.cfdi;

import com.business.unknow.Constants.CfdiConstants;
import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;
import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlAttribute;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlElementWrapper;
import javax.xml.bind.annotation.XmlRootElement;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@XmlRootElement(name = "Comprobante", namespace = "http://www.sat.gob.mx/cfd/3")
@XmlAccessorType(XmlAccessType.NONE)
@SuperBuilder(toBuilder = true)
@AllArgsConstructor()
@Setter
@Getter
public class Cfdi {

  @XmlAttribute(name = "Version")
  private String version = CfdiConstants.FACTURA_VERSION;

  @XmlAttribute(name = "Serie")
  private String serie;

  @XmlAttribute(name = "Folio")
  private String folio;

  @XmlAttribute(name = "Fecha")
  private String fecha;

  @XmlAttribute(name = "Sello")
  private String sello;

  @XmlAttribute(name = "FormaPago")
  private String formaPago;

  @XmlAttribute(name = "NoCertificado")
  private String noCertificado;

  @XmlAttribute(name = "Certificado")
  private String certificado;

  @XmlAttribute(name = "SubTotal")
  private BigDecimal subtotal;

  @XmlAttribute(name = "Descuento")
  private BigDecimal descuento;

  @XmlAttribute(name = "TipoCambio")
  private BigDecimal tipoCambio;

  @XmlAttribute(name = "Moneda")
  private String moneda;

  @XmlAttribute(name = "Total")
  private BigDecimal total;

  @XmlAttribute(name = "TipoDeComprobante")
  private String tipoDeComprobante;

  @XmlAttribute(name = "MetodoPago")
  private String metodoPago;

  @XmlAttribute(name = "LugarExpedicion")
  private String lugarExpedicion;

  @XmlElement(name = "CfdiRelacionados", namespace = "http://www.sat.gob.mx/cfd/3")
  private CfdiRelacionados cfdiRelacionados;

  @XmlElement(name = "Emisor", namespace = "http://www.sat.gob.mx/cfd/3")
  private Emisor emisor;

  @XmlElement(name = "Receptor", namespace = "http://www.sat.gob.mx/cfd/3")
  private Receptor receptor;

  @XmlElementWrapper(name = "Conceptos", namespace = "http://www.sat.gob.mx/cfd/3")
  @XmlElement(name = "Concepto", namespace = "http://www.sat.gob.mx/cfd/3")
  private List<Concepto> conceptos;

  @XmlElement(name = "Impuestos", namespace = "http://www.sat.gob.mx/cfd/3")
  private Impuesto impuestos;

  @XmlElement(name = "Complemento", namespace = "http://www.sat.gob.mx/cfd/3")
  private Complemento complemento;

  public Cfdi() {
    this.conceptos = new ArrayList<>();
    this.impuestos = new Impuesto();
  }
}
